python/ray/dag/compiled_dag_node.py

```diff
@@ -386,6 +324,79 @@ def __init__(
             assert not isinstance(val, ChannelInterface)
             assert not isinstance(val, DAGInputAdapter)
 
+        self.input_reader: ReaderInterface = SynchronousReader(self.input_channels)
```

Comment:
I don't fully understand the question. Would you mind explaining it further? Thanks!

Reply:
we can handle it later. this just assumes reader/writer is syncrhonous, and I wondered if this should be passed as an input (so that we can support different implementation)